diff --git a/src/dns_client/client_mdns.c b/src/dns_client/client_mdns.c
index d32d7f9..5a550ff 100644
--- a/src/dns_client/client_mdns.c
+++ b/src/dns_client/client_mdns.c
@@ -170,8 +170,13 @@ int _dns_client_add_mdns_server(void)
 			continue;
 		}
 
-		addr = (const unsigned char *)&((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;
-		addr_len = sizeof(struct in_addr);
+		if (ifa->ifa_addr->sa_family == AF_INET6) {
+			addr = (const unsigned char *)&((struct sockaddr_in6 *)ifa->ifa_addr)->sin6_addr;
+			addr_len = sizeof(struct in6_addr);
+		} else {
+			addr = (const unsigned char *)&((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;
+			addr_len = sizeof(struct in_addr);
+		}
 
 		// Skip the local interface
 		if (strcmp(ifa->ifa_name, "lo") == 0 || strcmp(ifa->ifa_name, "localhost") == 0) {
